### DISCUSSION TITLE
Navier Stokes

### DISCUSSION THREAD

**Discussion by tonyzahtila** [NVIDIA/warp#68] [Q&A]
Hi, 

I am just wondering if the NVIDIA/warp library would be appropriate for a Navier Stokes solver? Such a solver would require FFTs performed across multiple GPUs. I can see in the examples that most of the functionality required is present but not FFTs.

Kind regards.

**Comment by mmacklin** [comment]
Hi @tonyzahtila ,

There is an example of a 2D Semi-Lagrangian solver in https://github.com/NVIDIA/warp/blob/main/examples/example_fluid.py.

This uses an iterative method (Jacobi) to solve the Poisson problem. You could swap this out for an FFT through another library like CuPy, though we may offer FFT as a built-in in the future.

Cheers,
Miles
